src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.wasm.cs

```diff
@@ -68,21 +68,35 @@ private void ConditionalUpdate(ref bool condition, Action action)
 
 		private void SynchronizeHtmlParagraphAttributes()
 		{
-			ConditionalUpdate(ref _fontStyleChanged, () => this.SetFontStyle(FontStyle));
-			ConditionalUpdate(ref _fontWeightChanged, () => this.SetFontWeight(FontWeight));
-			ConditionalUpdate(ref _fontFamilyChanged, () => this.SetFontFamily(FontFamily));
-			ConditionalUpdate(ref _fontSizeChanged, () => this.SetFontSize(FontSize));
-			ConditionalUpdate(ref _maxLinesChanged, () => this.SetMaxLines(MaxLines));
-			ConditionalUpdate(ref _textAlignmentChanged, () => this.SetTextAlignment(TextAlignment));
-			ConditionalUpdate(ref _lineHeightChanged, () => this.SetLineHeight(LineHeight));
-			ConditionalUpdate(ref _characterSpacingChanged, () => this.SetCharacterSpacing(CharacterSpacing));
-			ConditionalUpdate(ref _textDecorationsChanged, () => this.SetTextDecorations(TextDecorations));
-			ConditionalUpdate(ref _paddingChangedChanged, () => this.SetTextPadding(Padding));
-
-			if(_textTrimmingChanged || _textWrappingChanged)
+			// SetXXX doesn't do anything when a non-null list is passed as second argument.
+			// This avoids many JS calls by first collecting the desired values in the two lists.
+			var setStyles = new List<(string, string)>();
+			var resetStyles = new List<string>();
+			ConditionalUpdate(ref _fontStyleChanged, () => this.SetFontStyle(FontStyle, setStyles, resetStyles));
+			ConditionalUpdate(ref _fontWeightChanged, () => this.SetFontWeight(FontWeight, setStyles, resetStyles));
+			ConditionalUpdate(ref _fontFamilyChanged, () => this.SetFontFamily(FontFamily, setStyles, resetStyles));
+			ConditionalUpdate(ref _fontSizeChanged, () => this.SetFontSize(FontSize, setStyles, resetStyles));
+			ConditionalUpdate(ref _maxLinesChanged, () => this.SetMaxLines(MaxLines, setStyles, resetStyles));
+			ConditionalUpdate(ref _textAlignmentChanged, () => this.SetTextAlignment(TextAlignment, setStyles, resetStyles));
+			ConditionalUpdate(ref _lineHeightChanged, () => this.SetLineHeight(LineHeight, setStyles, resetStyles));
+			ConditionalUpdate(ref _characterSpacingChanged, () => this.SetCharacterSpacing(CharacterSpacing, setStyles, resetStyles));
+			ConditionalUpdate(ref _textDecorationsChanged, () => this.SetTextDecorations(TextDecorations, setStyles, resetStyles));
+			ConditionalUpdate(ref _paddingChangedChanged, () => this.SetTextPadding(Padding, setStyles, resetStyles));
```

Comment:
This is looking interesting! The only part that I think can be optimized is the closure on `setStyles` and `resetStyles`, which forces the instanciation of a display class. I think those two should be class members instead.

/cc @carldebilly

Reply:
> I think those two should be class members instead.

@jeromelaban Are you referring to `setStyles` and `resetStyles` lists? They're not needed anywhere else. What would be the benefit of having them as class members? Reduce allocations of new lists by clearing the existing ones probably?

Reply:
It'll avoid the compiler to create a closure over those two members (and a display class as result). Per your last change, making those class members is not needed anymore for that specific reason (as you removed the lambdas as well). Otherwise, it's looking good.